src/Sparrow/Collections/FastStack.cs

```diff
@@ -170,16 +207,13 @@ public bool TryPeek(int depth, out T result)
         public T Pop()
         {
             if (_size == 0)
-                goto Error;
+                throw new InvalidOperationException("The stack is empty.");
```

Comment:
So explicitly having `throw` is _always_ fine, or only in some specific cases? Should we still continue to follow `ThrowXXXX()` decorated by `[DoesNotReturn]` pattern where is the actual `throw`?

Reply:
It is a fine line. That would depend how compact is the method and therefore how much 'relative' time the JIT will have to analyze it. In this case what I wanted to do is remove the `goto` because it makes things difficultier for the Tier 1 JIT to get better code when inlining. In .Net 8.0 many of the restrictions are being lifted, but not all of them. At this level, it is all  a case by case. The safe approach is use the `ThrowXXXX` pattern still, but under certain conditions we may need to use something different.

Reply:
Thanks